landoapi/landing_worker.py

```diff
@@ -521,4 +522,17 @@ def run_job(
                 # the landing user so they are aware and can update the bugs themselves.
                 self.notify_user_of_bug_update_failure(job, e)
 
+        try:
+            # Tell Phabricator to scan the landing repo so revisions are closed quickly.
+            phab = PhabricatorClient(
+                current_app.config["PHABRICATOR_URL"],
+                current_app.config["PHABRICATOR_UNPRIVILEGED_API_KEY"],
+            )
+            phab.call_conduit("diffusion.looksoon", repositories=[repo.phab_identifier])
+
+        except PhabricatorAPIException as e:
+            # Log the exception but continue gracefully.
+            logger.exception("Failed calling `diffusion.looksoon` to update Phab.")
+            logger.exception(e)
+
```

Comment:
Couple of things to note here:

- This will be triggered quite frequently (maybe more than needed) when the landing worker is under a heavy load. What kind of performance impact is it going to have on Phabricator?
- This should be moved to an async command (similar to `admin_remove_phab_project`) so that it doesn't interfere with the Landing Worker
- Perhaps we can throttle the task somehow to make sure it isn't called after every single landing.

Reply:
> Couple of things to note here:
> 
>     * This will be triggered quite frequently (maybe more than needed) when the landing worker is under a heavy load. What kind of performance impact is it going to have on Phabricator?

Very little - the call itself is lightweight as it just passes the callsign to Phab, and Phab schedules a repo update sooner than the standard update frequency.

>     * This should be moved to an async command (similar to `admin_remove_phab_project`) so that it doesn't interfere with the Landing Worker

Makes sense, I'll do that.

>     * Perhaps we can throttle the task somehow to make sure it isn't called after every single landing.

I think that's mostly the point, we want the repo to update ASAP after landings take place. Do we know how long it usually takes the landing worker to complete each push? I would assume the processing time for each job would be long enough that hitting Phab with one of these requests wouldn't have a huge impact on performance.

Reply:
> Very little - the call itself is lightweight as it just passes the callsign to Phab, and Phab schedules a repo update sooner than the standard update frequency.

I think as long as Phabricator can handle these calls and they don't cause any problems from the phabricator daemon perspective, this makes sense to me.

> I think that's mostly the point, we want the repo to update ASAP after landings take place. Do we know how long it usually takes the landing worker to complete each push? I would assume the processing time for each job would be long enough that hitting Phab with one of these requests wouldn't have a huge impact on performance.

Given the above, this may be OK. We do store the job duration on each job in the DB so that can be checked, but it's definitely in the multiple seconds at the least.